### PR TITLE
Added method to remove unused materials / elements / isotopes

### PR DESCRIPTION
**Pull request by lobis**
https://github.com/mipt-npm/gdml.kt/issues/41

Code should work but it is not tested with geometry where material is composed of other materials (only tested when material is composed of multiple elements).

There are a few assertions commented in the test because I couldn't get them to work but if implemented they should provide a decent check.

I also updated the `babyIAXO.gdml` geometry for a new one with excess materials, in order to use in the test.

### CODE REVIEW

**Comment by altavir** [comment]
Thanks a lot. I will review and merge soon.

File: src/commonMain/kotlin/space/kscience/gdml/GdmlMaterialPostProcessor.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package space.kscience.gdml
+
+public class GdmlMaterialPostProcessor {
```
Comment (altavir) [inline src/commonMain/kotlin/space/kscience/gdml/GdmlMaterialPostProcessor.kt:3]: Make it `object GdmkMaterialPostProcessor` and move everything from companion to the object body.

Comment (altavir) [reply src/commonMain/kotlin/space/kscience/gdml/GdmlMaterialPostProcessor.kt:3]: Even better make it `fun Gdml.removeUnusedMaterials()` extension.

**Review by altavir — CHANGES_REQUESTED** [review]

File: src/commonMain/kotlin/space/kscience/gdml/GdmlMaterialPostProcessor.kt (hunk 1 above)
Comment (lobis) [reply src/commonMain/kotlin/space/kscience/gdml/GdmlMaterialPostProcessor.kt]: I have implemented the changes.

**Review by altavir — APPROVED** [review]

**Pull request merged**